docs/proposals/perf.md

```diff
@@ -0,0 +1,381 @@
+
+  Edge Nodes numbers are one of `[1, 10, 20, 50, 100, 200...]`.
+
+  Device numbers per Edge Node are one of `[1, 10, 20, 50, 100, 200...]`.
+
+These test cases should be run in both system idle and under heavy load.
+
+#### 6. Add Pod from CloudHub to EdgeHub
+<img src="../images/perf/perf-cloudhub-edgehub.png">
+
+This scenario is expected to measure the performance of KubeEdge between CloudHub to EdgeHub.
+Actually this is not an E2E Test scenario for KubeEdge,
+but the message delivery channel between CloudHub to EdgeHub may be our bottleneck.
+Currently we are using web socket as the communication protocal between Cloud and Edge.
+In the following test cases, we need to mock the behaviors of CloudHub and EdgeHub,
+and the simulation messages of adding pod will be sent to EdgeHub,
+and the simulation messages of pod status will be sent back to CloudHub.
+so that we can get the exact latency and throughput between CloudHub and EdgeHub.
+
+Test Cases:
+* Measure the latency between KubeEdge CloudHub and KubeEdge EdgeHub.
+
+  Edge Nodes numbers are one of `[1, 10, 20, 50, 100, 200...]`.
+
+  Pods numbers per Edge Node are one of `[1, 2, 5, 10, 20...]`.
+
+* Measure the throughput between KubeEdge CloudHub and KubeEdge EdgeHub.
+
+  Edge Nodes numbers are one of `[1, 10, 20, 50, 100, 200...]`.
+
+  Pods numbers per Edge Node are one of `[1, 2, 5, 10, 20...]`.
+
+* Measure CPU and Memory Usage of KubeEdge Cloud Part.
+
+  Edge Nodes numbers are one of `[1, 10, 20, 50, 100, 200...]`.
+
+  Pods numbers per Edge Node are one of `[1, 2, 5, 10, 20...]`.
+
+* Measure CPU and Memory Usage of KubeEdge Edge Part.
+
+  Edge Nodes numbers are one of `[1, 10, 20, 50, 100, 200...]`.
+
+  Pods numbers per Edge Node are one of `[1, 2, 5, 10, 20...]`.
+
+As the result of the latency and throughput, we can evaluate scalability of KubeEdge EdgeHubs also the same with KubeEdge Edge Nodes.
+
+## Thresholds
+
+As the result of Performance Test, we expect to determine the performance and scalability for KubeEdge.
+This is critical to make some improvement items for KubeEdge.
+On the other hand, it will give the users recommended setup and user guides on KubeEdge.
+
+As [this K8S document](https://github.com/kubernetes/community/blob/master/sig-scalability/configs-and-limits/thresholds.md) mentions,
+since 1.6 release K8S can support 5000 Nodes and 150000 Pods in single cluster.
+KubeEdge is based on K8S Master, the difference between K8S Nodes and KubeEdge Edge Nodes
+is that KubeEdge Edge Nodes are not directly connected with the K8S Master like the K8S Nodes do.
+The KubeEdge Cloud Part connects K8S Master with KubeEdge Edge Nodes.
+Certainly the KubeEdge Edge Nodes are light weight and making use of less resources like CPU, Memory.
+
+Currently about KubeEdge we have no performance data which can make comparison with the other systems.
+But we can measure the performance and scalability for KubeEdge by the Performance Test data.
```

Comment:
by the Performance Test data -> using the Performance Test data.

Reply:
Thanks. will update later